### PR TITLE
Feature/update to polkadot v0 9 16

### PR DESCRIPTION
**Pull request by MRamanenkau**

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -131,6 +132,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	impl_version: 0,
 	apis: RUNTIME_API_VERSIONS,
 	transaction_version: 2,
+	state_version: 1,
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs]: [Version of the state implementation used by this runtime](https://github.com/paritytech/substrate/pull/9732)

File: bin/node/runtime/src/lib.rs [hunk 2]
```diff
@@ -898,10 +918,8 @@ impl pallet_tips::Config for Runtime {
 }
 
 parameter_types! {
-	pub ContractDeposit: Balance = deposit(
-		1,
-		<pallet_contracts::Pallet<Runtime>>::contract_info_size(),
-	);
+	pub const DepositPerItem: Balance = deposit(1, 0);
+	pub const DepositPerByte: Balance = deposit(0, 1);
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:903]: "Calling a contract will transfer balance from the caller to the contract or vise versa depending on whether the call increased or decreased the storage usage"
https://github.com/paritytech/substrate/pull/10082

File: bin/node/runtime/src/lib.rs [hunk 3]
```diff
@@ -132,7 +132,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	impl_version: 0,
 	apis: RUNTIME_API_VERSIONS,
 	transaction_version: 2,
-	state_version: 1,
+	state_version: 0,
```
Comment (AndreiNavoichyk) [inline bin/node/runtime/src/lib.rs:135]: Why `0`?

File: bin/node/runtime/src/lib.rs [hunk 4]
```diff
@@ -884,6 +876,15 @@ impl pallet_bounties::Config for Runtime {
 	type DataDepositPerByte = DataDepositPerByte;
 	type MaximumReasonLength = MaximumReasonLength;
 	type WeightInfo = pallet_bounties::weights::SubstrateWeight<Runtime>;
+	type ChildBountyManager = ChildBounties;
+}
+
+impl pallet_child_bounties::Config for Runtime {
```
Comment (AndreiNavoichyk) [inline bin/node/runtime/src/lib.rs:882]: Why do we need this pallet?

Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:882]: We use `pallet-bounties` and `pallet_child_bounties` [tightly coupled](https://paritytech.github.io/substrate/master/pallet_child_bounties/index.html) with it.

File: bin/node/runtime/src/lib.rs (hunk 3 above)
Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:135]: Update from 0 to 1 put the state in a hybride mode until we execute a migration. For more details check `pallet_state_trie_migration` in our [notion](https://www.notion.so/cere/Testing-logs-1f7d5107060a432398bad8b6a15225f0#8484d897223540f6af64fe918df33d4c)

Comment (AndreiNavoichyk) [reply bin/node/runtime/src/lib.rs:135]: I checked. For now we do not apply it and this is why we can stay with `state_version` 0. Correct?

**Review by AndreiNavoichyk — DISMISSED** [review]

File: bin/node/runtime/src/lib.rs (hunk 3 above)
Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:135]: Correct

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**